clang/lib/CodeGen/CGExpr.cpp

```diff
@@ -4051,6 +4141,14 @@ void CodeGenFunction::EmitTrapCheck(llvm::Value *Checked,
 
   llvm::BasicBlock *&TrapBB = TrapBBs[CheckHandlerID];
 
+  llvm::DILocation *TrapLocation = Builder.getCurrentDebugLocation();
+  llvm::StringRef TrapMessage = GetUBSanTrapForHandler(CheckHandlerID);
+
+  if (getDebugInfo() && !TrapMessage.empty()) {
+    TrapLocation = getDebugInfo()->CreateTrapFailureMessageFor(
```

Comment:
Lets add an `assert (TrapLocation && "Artificial trap frame requires valid debug-info for its scope.")`

So if we run into the crash again this will be easier to spot